### ISSUE TITLE
Enhancement: Add per-image reduction option to VisualInformationFidelity

### ISSUE THREAD

**Issue by iamkulbhushansingh** [Lightning-AI/torchmetrics#3194]
### Motivation

The current implementation of `VisualInformationFidelity (VIF)` returns a single scalar for an entire batch of images.  
This behavior makes it difficult to analyze per-image quality when dealing with multiple images.

Many other TorchMetrics metrics, such as `SSIM` and `PSNR`, already support a `reduction='none'` argument, returning per-image values.  
This enhancement would bring VIF in line with the design of these metrics.

### Pitch

- Add a `reduction` argument (`mean` | `none`) to `VisualInformationFidelity`.
- When `reduction='none'`, the metric should return a tensor of shape `(N,)` with per-image VIF values.
- Default behavior remains `mean` for backward compatibility.

## Example Usage
```python
from torchmetrics.image import VisualInformationFidelity
import torch

preds = torch.rand(4, 3, 256, 256)
target = torch.rand(4, 3, 256, 256)

metric = VisualInformationFidelity(reduction='none')
print(metric(preds, target))  # Expected output: tensor of shape (4,)
